Fix `NullReferenceException`: `R3Extensions.ApplyChanges[TKey,TValue]`

```
00:05:11.054 [ERROR] (NexusMods.App.Startup) Exception from Observable in onNext|System.NullReferenceException: Object reference not set to an instance of an object.
   at NexusMods.Abstractions.UI.Extensions.R3Extensions.ApplyChanges[TKey,TValue](IDictionary`2 dict, IChangeSet`2 changes) in C:\Dev\Repos\NexusMods.App\src\Abstractions\NexusMods.Abstractions.UI\Extensions\R3Extensions.cs:line 133
   at NexusMods.App.UI.Pages.LibraryPage.IHasLinkedLoadoutItems.<>c__DisplayClass4_0`1.<SetupLinkedLoadoutItems>b__1(IChangeSet`2 changes) in C:\Dev\Repos\NexusMods.App\src\NexusMods.App.UI\Pages\LibraryPage\ILibraryItemModel.cs:line 47
   at NexusMods.App.UI.ReactiveUiExtensions.ExceptionObserverWithOnNext`1.OnNextCore(T value) in C:\Dev\Repos\NexusMods.App\src\NexusMods.App.UI\ReactiveUIExtensions.cs:line 185
```
## Steps:
Install a new collection, or remove a collection. In general it seems to trigger when large amounts of changes get pushed quickly.

--- a/src/NexusMods.App.UI/Controls/TreeDataGrid/TreeDataGridAdapter.cs
+++ b/src/NexusMods.App.UI/Controls/TreeDataGrid/TreeDataGridAdapter.cs
@@ -7,6 +7,7 @@ using NexusMods.Abstractions.UI.Extensions;
 using NexusMods.App.UI.Extensions;
 using ObservableCollections;
 using R3;
+using System.Reactive.Linq;
 
 namespace NexusMods.App.UI.Controls;
 
@@ -99,8 +100,9 @@ public abstract class TreeDataGridAdapter<TModel, TKey> : ReactiveR3Object
                     return self
                         .GetRootsObservable(viewHierarchical)
                         .OnUI()
+                        .Do(changeSet => self.Roots.ApplyChanges(changeSet))
+                        .DisposeMany()
                         .ToObservable()
-                        .Do(self, static (changeSet, self) => self.Roots.ApplyChanges(changeSet))
                         .Select(viewHierarchical, static (_, viewHierarchical) => viewHierarchical);
                 })
                 .Switch()
@@ -118,7 +120,7 @@ public abstract class TreeDataGridAdapter<TModel, TKey> : ReactiveR3Object
             SingleSelect = false,
         };
 
-        var selectionObservable = Observable.FromEventHandler<TreeSelectionModelSelectionChangedEventArgs<TModel>>(
+        var selectionObservable = R3.Observable.FromEventHandler<TreeSelectionModelSelectionChangedEventArgs<TModel>>(
             addHandler: handler => selection.SelectionChanged += handler,
             removeHandler: handler => selection.SelectionChanged -= handler
         ).Select(tuple => tuple.e);

--- a/src/NexusMods.App.UI/Controls/TreeDataGrid/TreeDataGridAdapter.cs
+++ b/src/NexusMods.App.UI/Controls/TreeDataGrid/TreeDataGridAdapter.cs
@@ -99,7 +99,6 @@ public abstract class TreeDataGridAdapter<TModel, TKey> : ReactiveR3Object
                     return self
                         .GetRootsObservable(viewHierarchical)
                         .OnUI()
-                        .DisposeMany()
                         .ToObservable()
                         .Do(self, static (changeSet, self) => self.Roots.ApplyChanges(changeSet))
                         .Select(viewHierarchical, static (_, viewHierarchical) => viewHierarchical);

--- a/src/NexusMods.App.UI/Pages/LibraryPage/NexusModsModPageLibraryItemModel.cs
+++ b/src/NexusMods.App.UI/Pages/LibraryPage/NexusModsModPageLibraryItemModel.cs
@@ -120,7 +120,8 @@ public class NexusModsModPageLibraryItemModel : TreeDataGridItemModel<ILibraryIt
             FormattedInstalledDate,
             InstallItemCommand,
             IsInstalled,
-            InstallButtonText
+            InstallButtonText,
+            linkedLoadoutItemsDisposable
         );
     }
 

